### DISCUSSION TITLE
Any Chance flashing AT32F403ACGU7 BlackPill

### DISCUSSION THREAD

**Discussion by basst22778** [shanteacontrols/OpenDeck#160] [Q&A]
Hi everyone, 
today my parcel from ali express arrived.  Unfortunately i ordered the wrong blackpill boards.

Is there any chance to build a binary for this board?
AT32F403ACGU7 - WeAct Studio

**Comment by paradajz** [comment]
If I'm not mistaken that's a clone of some STM32F3 MCU, so no, the firmware won't work with that board.

**Reply by basst22778** [reply]
that's sad. Because the board is linked in the readme file.
[STM32F411 Black Pill](https://stm32-base.org/boards/STM32F411CEU6-WeAct-Black-Pill-V2.0) ([Adafruit](https://www.adafruit.com/product/4877) / [AliExpress](https://www.aliexpress.com/item/1005001456186625.html?pdp_ext_f=%7B%22sku_id%22:%2212000016817645221%22,%22ship_from%22:%22CN%22%7D&gps-id=pcStoreJustForYou&scm=1007.23125.137358.0&scm_id=1007.23125.137358.0&scm-url=1007.23125.137358.0&pvid=c145b0d6-10a9-42f7-9ddf-a3701e2ee9a1&spm=a2g0o.store_pc_home.smartJustForYou_6000147819213.1))

**Reply by paradajz** [reply]
They've changed the MCU due to the chip shortage. There's a note on their page:

> Information Note
> 
> Since STM32F411CEU6 chip is too expensive, due to the cost and customer demand. 
> We decided to use AT32F403ACGU7 as the new version of blackbill, with higher performance, larger memory and more affordable price.

**Reply by basst22778** [reply]
Too bad, now I have 4 of them lying around. If you can use some of them, I can send them to you.

**Reply by paradajz** [reply]
I don't need them, but thanks. I'm sorry for the wasted purchase! For what it's worth, I've explicitly mentioned in the [README](https://github.com/shanteacontrols/OpenDeck/blob/master/README.md) that AT32F403ACGU7 isn't supported and removed the link from their store.

**Comment by paradajz** [comment]
In case you're interested, I've just added support for Raspberry Pi Pico and Pico variant from [WeAct](https://www.aliexpress.com/item/1005003708090298.html?pdp_ext_f=%7B%22sku_id%22:%2212000029284764160%22,%22ship_from%22:%22CN%22%7D&gps-id=pcStoreLeaderboard&scm=1007.22922.271278.0&scm_id=1007.22922.271278.0&scm-url=1007.22922.271278.0&pvid=41bf8177-c3f6-4f3b-9f1f-ba67a2e46422&spm=a2g0o.store_pc_home.smartLeaderboard_6000147819209.1005003708090298) as well.

**Reply by AlpesMachines** [reply]
ahhhh that's a good news ! do you provide possibility to generate a midi beat clock in order to implement a sequencer beside the midi controls ?

**Reply by paradajz** [reply]
No, any sequencer-related functionality isn't on the roadmap.

**Reply by gh0stless** [reply]
A MIDI Clock Generator is also my biggest wish, all what need is a free timer?

**Reply by basst22778** [reply]
> A MIDI Clock Generator is also my biggest wish, all what need is a free timer?

Just take a look at pedalino mini. It is also Open Source and based on ESP32. They have limited inputs (6) but Midi Clock implememtation
